src/fstext/lattice-weight.h

```diff
@@ -330,6 +330,34 @@ class NaturalLess<LatticeWeightTpl<FloatType> > {
     return (Compare(w1, w2) == 1);
   }
 };
+template<>
+class NaturalLess<LatticeWeightTpl<float> > {
+ public:
+  typedef LatticeWeightTpl<float> Weight;
```

Comment:
Just wondering, did `LatticeWeightTpl` break for you too? I tried only cl and icc on Windows so far, and they only complained about the 2-argument template `CompactLatticeWeightTpl`. But the both compilers are minimum C++14, I think. A 1-argument specialization is "more specific" in C++, so both specialization should be fine, if you bake in the `int32` as the Gallic label type into `CompactLatticeWeightTpl`.

Reply:
Yes it broke for us, that's why we changed this code.  Kaldi only requires c++11 to compile, so we don't want to depend on c++14 features.